src/core/annotation.js

```diff
@@ -624,11 +624,17 @@ class WidgetAnnotation extends Annotation {
 
     data.readOnly = this.hasFieldFlag(AnnotationFieldFlag.READONLY);
 
-    // Hide signatures because we cannot validate them, and unset the fieldValue
-    // since it's (most likely) a `Dict` which is non-serializable and will thus
-    // cause errors when sending annotations to the main-thread (issue 10347).
+    // Hide signatures because we cannot validate them, and replace fieldValue
+    // with the signature, the content range, and signer's name only, since it's
+    // (most likely) a `Dict` which is non-serializable and will thus cause
+    // errors when sending annotations to the main-thread (issue 10347).
     if (data.fieldType === 'Sig') {
-      data.fieldValue = null;
+      data.fieldValue = {
+        ByteRange: data.fieldValue.get('ByteRange'),
+        Contents: data.fieldValue.get('Contents'),
+        Name: data.fieldValue.get('Name'),
```

Comment:
In a malformed PDF document, there's nothing stopping any of the properties above from being e.g. a `Dict` or a `Stream`, thus making them non-serializable which will throw errors during `postMessage`.

Assuming that we actually want to do this, which seems questionable since this is essentially adding dead code which normally isn't accepted (there's no call-site/unit-test using this information), there would need to be *a lot* more validation added for this to be even remotely safe.

Reply:
> In a malformed PDF document, there's nothing stopping any of the properties above from being e.g. a `Dict` or a `Stream`, thus making them non-serializable which will throw errors during `postMessage`.

Oh, good point.

> Assuming that we actually want to do this, which seems questionable since this is essentially adding dead code

Totally agree. As I said, don't think this is useful at all for the project, but instead something I need for a project.

I've been looking around, but haven't found any way to get the data I need out of the worker. Is there something I'm missing? — It seems the main thread is prohibited from touching the PDF outside of the AST the worker offers.